astroquery/sdss/core.py

```diff
@@ -94,76 +103,86 @@ def query_crossid_async(self, coordinates, obj_names=None,
+            Normally cross-id only returns the closest primary object.
+            Setting this to ``True`` will return all objects.
+        field_help: str or bool, optional
+            Field name to check whether a valid PhotoObjAll or SpecObjAll
+            field name. If `True` or it is an invalid field name all the valid
+            field names are returned as a dict.
+        get_query_payload : bool, optional
             If True, this will return the data the query would have sent out,
             but does not actually do the query.
-        data_release : int
+        data_release : int, optional
             The data release of the SDSS to use.
-        """
+        cache : bool, optional
+            If ``True`` use the request caching mechanism.
 
-        if (not isinstance(coordinates, list) and
-                not isinstance(coordinates, Column) and
-                not (isinstance(coordinates, commons.CoordClasses) and
-                     not coordinates.isscalar)):
-            coordinates = [coordinates]
+        Returns
+        -------
+        result : `~astropy.table.Table`
+            The result of the query as a `~astropy.table.Table` object.
 
-        if obj_names is None:
-            obj_names = ['obj_{0}'.format(i) for i in range(len(coordinates))]
-        elif len(obj_names) != len(coordinates):
-            raise ValueError("Number of coordinates and obj_names should "
-                             "be equal")
+        """
 
-        if isinstance(radius, u.Quantity):
+        if isinstance(radius, Angle):
             radius = radius.to(u.arcmin).value
         else:
             try:
-                float(radius)
-            except TypeError:
+                radius = Angle(radius).to(u.arcmin).value
+            except ValueError:
                 raise TypeError("radius should be either Quantity or "
                                 "convertible to float.")
+        if radius > self.MAX_CROSSID_RADIUS.value:
+            raise ValueError(f"radius must be less than {self.MAX_CROSSID_RADIUS}.")
 
-        sql_query = 'SELECT\r\n'  # Older versions expect the CRLF to be there.
-
-        if specobj_fields is None:
-            if photoobj_fields is None:
-                photoobj_fields = crossid_defs
-            photobj_fields = ['p.{0}'.format(i) for i in photoobj_fields]
-            photobj_fields.append('p.objID as obj_id')
-            specobj_fields = []
+        if (not isinstance(coordinates, list) and
+            not isinstance(coordinates, Column) and
+            not (isinstance(coordinates, commons.CoordClasses) and
+                 not coordinates.isscalar)):
+            coordinates = [coordinates]
+        if obj_names is None:
+            obj_names = ['obj_{0}'.format(i) for i in range(len(coordinates))]
+        elif len(obj_names) != len(coordinates):
+            raise ValueError("Number of coordinates and obj_names should "
+                         "be equal")
```

Comment:
I like very explicit error messages.
```suggestion
            raise ValueError(
                f"number of coordinates {len(coordinates)} is not equal to number of object names {len(obj_names)}"
            )
```

Reply:
I was trying to make a minimal change. My personal experience is that PR reviewers are more likely to say "don't change what isn't broken".

Reply:
If this is an already existing error message then it should be modified in a separate pull request.